core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1606,11 +1606,15 @@ class SparkContext(config: SparkConf) extends Logging {
     new ReliableCheckpointRDD[T](this, path)
   }
 
+  protected[spark] def isPartitionerAwareUnion[T: ClassTag](rdds: Seq[RDD[T]]): Boolean = {
```

Comment:
Could you add a comment about the assumption, `rdds.filter(!_.partitions.isEmpty)`?

Reply:
Added comment and a check.